tasks/password-generator/src/main.rs

```diff
@@ -0,0 +1,172 @@
+// declare our external dependencies
+// for parsing command line arguments
+extern crate clap;
+// for working with random values
+extern crate rand;
+// why use a library for this CLI? Clap lets us quickly build a CLI and
+// lets us focus the below code on password generation
+// not the intricacies building a CLI in rust.
+// Read more about clap here: https://clap.rs/
+// bring the needed structures into scope so we
+// invoke them later in the program
+use clap::{App, Arg};
+use rand::distributions::Alphanumeric;
+use rand::prelude::IteratorRandom;
+use rand::{thread_rng, Rng};
+use std::iter;
+
+// the core logic that creates our password
+fn generate_password(length: usize) -> String {
+    // cache thread_rng for better performance
+    let mut rng = thread_rng();
+    let mut base_password: Vec<char> = iter::repeat(())
+        // the Alphanumeric struct provides 3/4
+        // of the characters for passwords
+        // so we can sample from it
+        .map(|()| rng.sample(Alphanumeric))
+        .take(length)
+        .collect();
+    // create an iterator of required other characters
+    let other_values = "!\"#$%&'()*+,-./:;<=>?@[]^_{|}~";
+    // create a random count of how many other characters to add
+    let mut to_add = rng.gen_range(1, 10);
+    loop {
+        let special = other_values.chars().choose(&mut rng).unwrap();
+        to_add -= 1;
+        base_password[to_add] = special;
+        if to_add == 0 {
+            break;
+        }
+    }
+    // you convert the vector of characters into a string
+    // using the turbofish syntax
+    base_password.iter().collect::<String>()
+}
+// validator function for our first argument
+// picked 15 as the length must be greater than
+// our randomly generated range above (10)
+fn is_valid_length(value: String) -> Result<(), String> {
+    let length = value.parse::<u32>();
+    // if we cannot parse the string into an i32
+    if length.is_err() {
+        Err(String::from("Please provide valid password"))
```

Comment:
This error message should be clearer, but I think if you use structopt it should handle the parsing for you.